src/Task/Recurrence.ts

```diff
@@ -5,12 +5,30 @@ import type { Moment } from 'moment';
 import { RRule } from 'rrule';
 import { compareByDate } from '../lib/DateTools';
 
+export class Occurrence {
+    readonly startDate: Moment | null;
+    readonly scheduledDate: Moment | null;
+    readonly dueDate: Moment | null;
+
+    constructor({
+        startDate,
+        scheduledDate,
+        dueDate,
+    }: {
+        startDate: Moment | null;
+        scheduledDate: Moment | null;
+        dueDate: Moment | null;
+    }) {
+        this.startDate = startDate;
+        this.scheduledDate = scheduledDate;
+        this.dueDate = dueDate;
+    }
+}
```

Comment:
That's rather repetitive - What's the motivation behind using a destructured object for the constructor?